src/main/java/org/fcrepo/spec/testsuite/authn/AuthenticatorResolver.java

```diff
@@ -0,0 +1,182 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.spec.testsuite.authn;
+
+import static org.fcrepo.spec.testsuite.TestParameters.AUTHENTICATOR_CLASS_PARAM;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.IOCase;
+import org.apache.commons.io.filefilter.FileFilterUtils;
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.util.ClasspathHelper;
+import org.reflections.util.ConfigurationBuilder;
+
+/**
+ * @author dbernstein
+ */
+public class AuthenticatorResolver {
+
+    private static String authenticatorClass = null;
+
+    private AuthenticatorResolver() {
+    }
+
+    private static Authenticator instance = null;
+
+    /**
+     * Initialize the resolver
+     *
+     * @param authenticatorClazz Optionally specify the class name of the Authenticator implementation.
+     */
+    public static void initialize(final String authenticatorClazz) {
+        authenticatorClass = authenticatorClazz;
+        instance = findAuthenticator();
+    }
+
+    private static Authenticator findAuthenticator() {
+        try {
+            final List<ClassLoader> classLoadersList = new LinkedList<>();
+            final String path =
+                AuthenticatorResolver.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+            final File jar = new File(path);
+
+            final String authDirName = "authenticators";
+
+            //look for jars an authenticators directory sitting in the same
+            //directory as the executing jar.
+            File dir = new File(jar.getParentFile(), authDirName);
+            //if there is no authenticators dir in the executing jar's dir
+            //look for an authenticators dir in the dir where java was run
+            if (!dir.exists()) {
+                dir = new File(System.getProperty("user.dir"), authDirName);
+                if (!dir.exists()) {
+                    //otherwise look in the user's home directory
+                    dir = new File(System.getProperty("user.home"), authDirName);
+                    if (!dir.exists()) {
+                        dir = null;
+                    }
+                }
+            }
+
+            //if a valid directory was found, try to find an implementation of the Authenticator
+            if (dir != null) {
+                final List<File> jars =
+                    Arrays.asList(
+                        dir.listFiles((FilenameFilter) FileFilterUtils.suffixFileFilter(".jar", IOCase.INSENSITIVE)));
```

Comment:
My implementation is not found since it is a `.class` file.

Reply:
did you try jarring it?  Or do you feel that it should work if it is a .class file?